xmtp_mls/src/client.rs

```diff
@@ -607,7 +607,20 @@ where
                                 match result {
                                     Ok(mls_group) => Ok(Some(mls_group)),
                                     Err(err) => {
-                                        log::error!("failed to create group from welcome: {}", err);
+                                        if let GroupError::Storage(StorageError::DieselResult(diesel::result::Error::DatabaseError(
+                                            diesel::result::DatabaseErrorKind::UniqueViolation,
+                                            ref error_info,
+                                        ))) = err
+                                        {
```

Comment:
I feel like we should create a standalone function that accepts a `Result<T, Error>`, printing any kind of log messages for the error and returning `T` if no error. 

This way we don't need to burden our functions with these checks, and can re-use the function for other `UniqueViolations` or errors when needed

Also, we should have a comment on the function explaining that if diesel changes any of these error messages, then this message turns into a hard-error